### PR TITLE
Removed _isMemoryLimitReached() nonsense

### PR DESCRIPTION
**Pull request by woutersamaey**
### Description (*)
If we don't have enough memory, we'll know soon enough. No need to do these computations every time.
### Questions or comments
- Less time/resources wasted on calculating memory needed.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by luigifab — COMMENTED** [review]
There is also a `_checkMemory` in `Mage_Catalog_Model_Product_Image`. On our store we have added `return true;`.

**Comment by woutersamaey** [comment]
Good catch @luigifab, I removed these as well.

File: app/code/core/Mage/Catalog/Model/Product/Image.php [hunk 1]
```diff
@@ -244,84 +244,6 @@ public function setSize($size)
         return $this;
     }
 
-    /**
-     * @param null $file
-     * @return bool
-     */
-    protected function _checkMemory($file = null)
```
Comment (Flyingmana) [inline app/code/core/Mage/Catalog/Model/Product/Image.php:251]: also removing the function may break modules using it. Please only remove the usages, and optionally mark the method as deprecated.
Please also do not remove the functionality, yet.

**Review by Flyingmana — CHANGES_REQUESTED** [review]

**Comment by Flyingmana** [comment]
Generally support the removal of the check. The PHP way of reporting reaching the memory limit is actually better and more verbose.
Although, this check allows for a more gracefully fail, as an exception allows handling it, while the php way lets the script die if I remember correctly.

**Comment by woutersamaey** [comment]
@Flyingmana you are correct, but Magento does nothing with the exception that is worth our consideration.
Any script can run out of memory, so it's up to the developer to watch for memory issues in the PHP log.

**Comment by Flyingmana** [comment]
@woutersamaey yes, thats why I support the change :)
But we need to keep the protected functions, in case other modules/shops use them on purpose.

**Comment by woutersamaey** [comment]
@Flyingmana who in their right mind is going to use this on purpose?
I say we clean up a bit and move forward, not stay in place. This is the v20 branch, right?

**Comment by Flyingmana** [comment]
It is not ours to judge others decision, and I want to please ask you to not talk ill of others, especially  not knowing which circumstances might have lead to these decisions.

Allowing BC breaks in the v20 branch is not an excuse to act reckless in this regard.

**Comment by woutersamaey** [comment]
Okay, fair enough. I didn't want to be rude, just can't see what possible purpose it could serve.
Is OpenMage ever planning on breaking away or will things stay forever as they are? Would be nice to see a roadmap or something like that...

**Comment by fballiano** [comment]
I support the cleaning tasks for v20 (some would say too much ehehehe), in this case, seeing you already discussed it, I'd just remove the calls to the "checkmemory" methods and mark all related methods as deprecated, so we can move this PR forward as we're all wasting computing cycles now :-)

**Comment by elidrissidev** [comment]
I support this. Let's just remove the usages and mark the method as deprecated for the v19 branch.

**Review by tobihille — APPROVED** [review]

**Review by tmewes — DISMISSED** [review]

**Comment by tmewes** [comment]
@fballiano Can you please push the magic button (update the branch with the current version of 20.0) that we get rid of the conflicts and get this PR done? :)

**Comment by fballiano** [comment]
I've restored the original methods marking them as deprecated, so it should be ready to merge.

But, I can't switch this PR to 1.9.4.x cause it's breaking for some reasons (it generates 90+ commits and 5000 modified files), if you're ok with it I'd close this one and recreate it as a new PR on v19.

**Comment by tmewes** [comment]
Sounds like a good plan. 👍

**Comment by fballiano** [comment]
Closing this in favour of https://github.com/OpenMage/magento-lts/pull/2471 which is based on v19 and non-BC :-)

**Pull request closed without merging**